### PR TITLE
Enable `make markdown` and `make text` for document

### PR DESCRIPTION
**Pull request by HashidaTKS**
## Purpose

This changes enable to create markdown and text files used for #1489 with the `make` command.

## Usage

```
cd doc
make markdown
```

The command above generates markdown files into `doc/locale/${locale}/markdown` from `.rst` files.
Note that `make markdown` requires [sphinx-markdown-builder](https://github.com/clayrisser/sphinx-markdown-builder)

```
cd doc
make text
```

The command above generates text files into `doc/locale/${locale}/text` from `.rst` files.

```
cd doc
make clean-markdown
```

The command above remove `doc/locale/${locale}/markdown`

```
cd doc
make clean-text
```

The command above remove `doc/locale/${locale}/text`

### CODE REVIEW

**Comment by HashidaTKS** [comment]
when I execute `make text` on `doc/locale/en`, there are no rule to build `LC_MESSAGES/characteristic.po` and the build failes.
After executing `make html`, I can build text with `make text`.

```
~/gitdir/groonga/doc/locale/en$ make text
make: *** 'text-build-stamp' に必要なターゲット 'LC_MESSAGES/characteristic.po' を make するルールがありません.  中止.
```

On the other hand, when I execute `make text` on `doc/locale/ja`, it success without executing `make html`.
It seems odd...

File: doc/Makefile.am [hunk 1]
```diff
@@ -50,6 +54,11 @@ if DOCUMENT_BUILDABLE
 dist-hook: html
 	@touch $(distdir)/build-stamp
 
+text: text-recursive
+text-am:
+markdown: markdown-recursive
+markdown-am:
```
Comment (HashidaTKS) [inline doc/Makefile.am:60]: `make markdown` fails with "there is no rule to make markdown-am" like below if this target is omitted, so I have added this.

```
make[3]: ディレクトリ '/home/t-hashida/gitdir/groonga/doc/locale/en' に入ります
make[3]: *** ターゲット 'markdown-am' を make するルールがありません.  中止.
```

However, the `html` target don't define any task like `html-am` in `.am` files, but it is defined in `Makefile`s.
Who is define `html-am`? And it it good to define `markdown-am` and `text-am` manually in the `.am` files?

**Comment by HashidaTKS** [comment]
@kou @komainu8 

Would you review this when you have time?

File: doc/Makefile.am [hunk 2]
```diff
@@ -4,6 +4,10 @@ SUBDIRS =					\
 BUILT_SOURCES =
 CLEANFILES =
 
+am__extra_recursive_targets = 	\
+	text-recursive		\
+	markdown-recursive
```
[the remark this answers is not in the document]
Comment (HashidaTKS) [reply doc/Makefile.am]: Thanks, I have fixed to use `AM_EXTRA_RECURSIVE_TARGETS()`.

**Comment by HashidaTKS** [comment]
@kou

Thanks, I have addressed your comment.
Would you re-review this when you have time?

**Comment by kou** [comment]
It seems that `sphinx-markdown-builder` doesn't support `csv-table` well:

```markdown
| タグ

 | 登場回数

 |
| `Groonga`

 | 4

    |
| `Mroonga`

 | 2

    |
| `PGroonga`

 | 2

    |
| `Groonga` かつ `Mroonga`

 | 2

    |
| `Groonga` かつ `PGroonga`

 | 2

    |
| `Mroonga` かつ `PGroonga`

 | 1

    |
| `Groonga` かつ `Mroonga` かつ `PGroonga`

 | 1

    |
```

We may want to use Markdown for the original source format.

**Pull request merged**